crates/kitsune_p2p/kitsune_p2p/src/metrics.rs

```diff
@@ -115,16 +118,29 @@ pub struct NodeInfo {
     /// Running average for latency microseconds for any direct
     /// request/response calls to remote agent.
     pub latency_micros: RunAvg,
+    /// Times we recorded successful initiates to this node (they accepted).
+    pub initiates: VecDeque<RoundMetric>,
+    /// Times we recorded initates from this node (we accepted).
+    pub accepts: VecDeque<RoundMetric>,
+    /// Times we recorded complete rounds for this node.
+    pub successes: VecDeque<RoundMetric>,
     /// Times we recorded errors for this node.
     pub errors: VecDeque<RoundMetric>,
-    /// Times we recorded initiates to this node.
-    pub initiates: VecDeque<Instant>,
-    /// Times we recorded remote rounds from this node.
-    pub remote_rounds: VecDeque<Instant>,
-    /// Times we recorded complete rounds for this node.
-    pub complete_rounds: VecDeque<RoundMetric>,
     /// Is this node currently in an active round?
-    pub current_round: Option<RoundMetric>,
+    pub current_round: bool,
+}
+
+/// Detailed info about the history of gossip with this node
+#[derive(Debug, Clone, Default)]
+pub struct PeerNodeHistory {
+    /// The most recent list of remote agents reported by this node
+    pub remote_agents: Vec<Arc<KitsuneAgent>>,
+
+    /// Detailed info about the ongoing round with this node
+    pub current_round: Option<CurrentRound>,
+
+    /// Detailed info about rounds completed with this node
+    pub completed_rounds: VecDeque<CompletedRound>,
 }
```

Comment:
@neonphog here is the first instance of node-based as opposed to agent-based metrics